fe/fe-core/src/main/java/org/apache/doris/datasource/DataSourceMgr.java

```diff
@@ -95,6 +96,20 @@ private DataSourceIf removeCatalog(long catalogId) {
         return catalog;
     }
 
+    private void unprotectedRefreshCatalog(long catalogId) {
+        DataSourceIf catalog = idToCatalog.get(catalogId);
+        if (catalog != null) {
+            String catalogName = catalog.getName();
+            if (!catalogName.equals(InternalDataSource.INTERNAL_DS_NAME)) {
+                ((ExternalDataSource) catalog).setInitialized(false);
+            }
+            catalog = nameToCatalog.get(catalogName);
+            ((ExternalDataSource) catalog).setInitialized(false);
```

Comment:
The catalog save in `idToCatalog` and `nameToCatalog` is same object instance. So no need to set them twice.

Reply:
My fault. I'll fix it later.